apis/python/src/tiledb/vector_search/module.cc

```diff
@@ -95,66 +133,73 @@ static void declare_pyarray_to_matrix(py::module& m, const std::string& suffix)
           throw std::runtime_error("Mismatched buffer format!");
 
         auto data = std::unique_ptr<T[]>{new T[info.shape[0] * info.shape[1]]};
-        std::memcpy(data.get(), info.ptr, info.shape[0] * info.shape[1] * sizeof(T));
-        auto r = ColMajorMatrix<T>(std::move(data), info.shape[0], info.shape[1]);
+        std::memcpy(
+            data.get(), info.ptr, info.shape[0] * info.shape[1] * sizeof(T));
+        auto r =
+            ColMajorMatrix<T>(std::move(data), info.shape[0], info.shape[1]);
         return r;
-        });
+      });
 }
 
 namespace {
- template <typename ...TArgs>
- py::tuple make_python_pair(std::tuple<TArgs...>&& arg) {
-    static_assert(sizeof...(TArgs) == 2, "Must have exactly two arguments");
+template <class... TArgs>
+py::tuple make_python_pair(std::tuple<TArgs...>&& arg) {
+  static_assert(sizeof...(TArgs) == 2, "Must have exactly two arguments");
 
-   return py::make_tuple<py::return_value_policy::automatic>(
+  return py::make_tuple<py::return_value_policy::automatic>(
       py::cast(std::get<0>(arg), py::return_value_policy::move),
-      py::cast(std::get<1>(arg), py::return_value_policy::move)
-   );
- }
-
+      py::cast(std::get<1>(arg), py::return_value_policy::move));
 }
 
-template <typename T, typename Id_Type = uint64_t>
-static void declare_qv_query_heap_infinite_ram(py::module& m, const std::string& suffix) {
-  m.def(("qv_query_heap_infinite_ram_" + suffix).c_str(),
-      [](const ColMajorMatrix<T>& parts,
-         const ColMajorMatrix<float>& centroids,
+}  // namespace
+
+template <class T, class Id_Type = uint64_t>
+static void declare_qv_query_heap_infinite_ram(
+    py::module& m, const std::string& suffix) {
+  m.def(
+      ("qv_query_heap_infinite_ram_" + suffix).c_str(),
+      [](ColMajorMatrix<T>& parts,          // Note, will be moved!
+         ColMajorMatrix<float>& centroids,  // Note, will be moved!
          const ColMajorMatrix<float>& query_vectors,
          std::vector<Id_Type>& indices,
          std::vector<Id_Type>& ids,
          size_t nprobe,
          size_t k_nn,
-         size_t nthreads) -> py::tuple { //std::pair<ColMajorMatrix<float>, ColMajorMatrix<size_t>> { // TODO change return type
+         size_t nthreads) -> py::tuple {
+
+        auto mat = ColMajorPartitionedMatrixWrapper<T, Id_Type, Id_Type>(
+            parts, ids, indices);
+
+        auto top_centroids = detail::ivf::ivf_top_centroids(
+            centroids, query_vectors, nprobe, nthreads);
+        auto r = detail::ivf::qv_query_heap_infinite_ram(
+            top_centroids, mat, query_vectors, nprobe, k_nn, nthreads);
```

Comment:
what is the purpose of this change?

Reply:
The API for all of the query functions (`qv`, `nuv`, etc.) changed to take relevant centroids, the partitioned matrix (encapsulating partitioned vectors, partitioned ids, and the partitioning index), and the queries.  So the calls to the C++ query functions from Python required some setup to call via the new API.  This is also intended to be transitional since we will (presumably) be implementing the indexes in C++ at some point, in which case there will also be a more sensible Python API.